### DISCUSSION TITLE
wired table 2 模型，在用polygons切割单元格时，有一行多切割了，不知道是模型问题，还是代码问题？

### DISCUSSION THREAD

**Discussion by nissansz** [RapidAI/TableStructureRec#41] [Q&A]
![image](https://github.com/user-attachments/assets/5612f2b7-1f47-4cdf-a33e-403889cf98b8)

        try:
            table_res, logi_points = self.table_recover(polygons)
            # 将坐标由逆时针转为顺时针方向，后续处理与无线表格对齐
            polygons[:, 1, :], polygons[:, 3, :] = (
                polygons[:, 3, :].copy(),
                polygons[:, 1, :].copy(),
            )
            if ocr_result is None:
                for i in range(polygons.shape[0]):
                    crop_img = get_rotate_crop_image(img, polygons[i])

**Comment by Joker1212** [comment]
使用wired_v2是没有问题的，你可以试试更新一下包，因为之前更新过一次

**Reply by nissansz** [reply]
我就是用的v2，今天下载的
wired-table-rec                               1.1.2

**Reply by Joker1212** [reply]
多切割了是指？能贴图说明一下预期是什么，目前是什么嘛？

**Reply by nissansz** [reply]
一开始就已经贴图了。红色那行

**Reply by nissansz** [reply]
![table2](https://github.com/user-attachments/assets/ab48344e-2f16-42ae-9472-b8587049dfcd)
这是测试的原图

**Reply by Joker1212** [reply]
意思是把一个单元格内的两行，切割为了上下两个单元格？

**Reply by nissansz** [reply]
![image](https://github.com/user-attachments/assets/8bf9e41d-ed9a-4cab-a83b-851beffc7cce)
红线所在的四个单元格坐标下面的线有问题，切出来的比原来的小，最下面一行文字只剩一半

**Reply by Joker1212** [reply]
我在线demo尝试，发现识别框是对齐的，html渲染也是正确的啊？没有后续改动吗？或者加一点padding兼容一下呢？
<img width="858" alt="image" src="https://github.com/user-attachments/assets/2b9288b8-4e0b-4b84-b87d-fd046259057f">

**Reply by nissansz** [reply]
我是用本地代码，改main.py那个文件，直接显示的，你能不能本地试试
try:
        table_res, logi_points = self.table_recover(polygons)
        # 将坐标由逆时针转为顺时针方向，后续处理与无线表格对齐
        polygons[:, 1, :], polygons[:, 3, :] = (
            polygons[:, 3, :].copy(),
            polygons[:, 1, :].copy(),
        )
        if ocr_result is None:
            for i in range(polygons.shape[0]):
                crop_img = get_rotate_crop_image(img, polygons[i])

**Reply by nissansz** [reply]
我是用本地代码，改main.py那个文件，直接显示的，你能不能本地试试
try:
        table_res, logi_points = self.table_recover(polygons)
        # 将坐标由逆时针转为顺时针方向，后续处理与无线表格对齐
        polygons[:, 1, :], polygons[:, 3, :] = (
            polygons[:, 3, :].copy(),
            polygons[:, 1, :].copy(),
        )
        if ocr_result is None:
            for i in range(polygons.shape[0]):
                crop_img = get_rotate_crop_image(img, polygons[i])

**Reply by Joker1212** [reply]
因为get_rotate_crop_image是做的透视变换，可能会有crop的误差问题，建议直接使用cv2依靠原坐标来截图，或者在后面加上padding

**Reply by nissansz** [reply]
直接使用cv2依靠原坐标来截图，x1,x2, y1,y2坐标怎么从polygons获取准？要转换？

**Reply by nissansz** [reply]
html, elasp, polygons, logic_points, ocr_res = table_engine(img)
这里返回的polygons已经是透视变换过的吗了？ 用返回的polygons 画方框，也是被切多了
for idx, polygon in enumerate(polygons):
    x0, y0, x1, y1 = polygon[0], polygon[1], polygon[2], polygon[3]
    x0 = round(x0)
    y0 = round(y0)
    x1 = round(x1)
    y1 = round(y1)
    cv2.rectangle(img, (x0, y0), (x1, y1), (0, 0, 255), 1)

![image](https://github.com/user-attachments/assets/bd99086d-19f4-4917-ae7a-0393925658da)

**Reply by Joker1212** [reply]
建议加一点padding吧

**Reply by nissansz** [reply]
怎么加比较好？加多了，会把其他单元格的内容弄进来吧？

**Reply by Joker1212** [reply]
就5-10的像素应该就可以

**Reply by nissansz** [reply]
![ea503e6dd76200e14538bc945e4e495](https://github.com/user-attachments/assets/ea7b9591-4f50-497d-9046-589647c622f8)
wired 2模型获取的原始polygons（不做任何调整的那种），在哪里可以返回，我想试试画框。

上面是模型作者测试的结果，明显要好于rapid纠正后的结果
